create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -32,17 +47,50 @@ public ScopusHandler(S3Client s3Client) {
     }
 
     @Override
-    public String handleRequest(S3Event event, Context context) {
-        return attempt(() -> readFile(event))
-            .map(this::parseXmlFile)
-            .map(this::getDoi)
-            .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+    public CreatePublicationRequest handleRequest(S3Event event, Context context) {
+        CreatePublicationRequest request = attempt(() -> readFile(event))
+                .map(this::parseXmlFile)
+                .map(this::extractMetadata)
+                .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+        return request;
+    }
+
+    private CreatePublicationRequest extractMetadata(DocTp docTp) {
+        CreatePublicationRequest request = new CreatePublicationRequest();
+        EntityDescription entityDescription = new EntityDescription();
+        request.setEntityDescription(entityDescription);
+        Reference reference = new Reference();
+        entityDescription.setReference(reference);
+        reference.setPublicationContext(getPublicationContext(docTp));
+        return request;
     }
 
-    private String getDoi(DocTp docTp) {
-        var doi= docTp.getMeta().getDoi();
-        logger.info("The publication doi:"+doi);
-        return doi;
+    private PublicationContext getPublicationContext(DocTp docTp) {
```

Comment:
This method is quite long and should be broken into smaller pieces

Reply:
Suggestion:
```
     public static final PublicationContext EMPTY_PUBLICATION_CONTEXT = null;

    private PublicationContext getPublicationContext(DocTp docTp) {
        if (isJournal(docTp)) {
            attempt(() -> createUnconfirmedJournal(docTp))
                .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
        }
        return EMPTY_PUBLICATION_CONTEXT; 
    }

    private UnconfirmedJournal createUnconfirmedJournal(DocTp docTp) throws InvalidIssnException {
        var source = getSource(docTp);
        var sourceTitle = extractSourceTitle(source);
        var issnTpList = source.getIssn();
        var printIssn = findPrintIssn(issnTpList);
        var electronicIssn = findElectronicIssn(issnTpList);
        return new UnconfirmedJournal(sourceTitle, printIssn, electronicIssn);
    }

```

Now there is the question, do we want to have a publication with an empty publication context?